operator/src/test/java/org/bf2/operator/managers/ImagePullSecretManagerTest.java

```diff
@@ -120,4 +124,66 @@ void testWithSecrets() {
                         .collect(Collectors.toSet()));
     }
 
+    @Test
+    void testPullSecretTypeChangePropagatedAsNewSecret() {
+        String ns = "testPullSecretTypeChangePropagatedAsNewSecret";
+        Secret originalNamespacedSecret = client.secrets()
+            .inNamespace(ns)
+            .create(new SecretBuilder()
+                    .withNewMetadata()
+                        .withName("mk-pull-name")
+                    .endMetadata()
+                    .withType("kubernetes.io/dockercfg")
+                    .withData(Map.of("key", "value"))
+                    .build());
+
+        Secret replacement = new SecretBuilder()
+            .withNewMetadata()
+                .withName("name")
+            .endMetadata()
+            .withType("kubernetes.io/dockerconfigjson")
+            .withData(Map.of("key", "value"))
+            .build();
+
+        ManagedKafka managedKafka = new ManagedKafka();
+        managedKafka.setMetadata(new ObjectMetaBuilder().withName("mk").withNamespace(ns).build());
+
+        imagePullSecretManager.propagateSecrets(client, managedKafka, List.of(replacement));
+
+        Secret replacementNamespacedSecret = client.secrets().inNamespace(ns).withName("mk-pull-name").get();
+
+        assertNotEquals(originalNamespacedSecret.getMetadata().getUid(), replacementNamespacedSecret.getMetadata().getUid());
```

Comment:
I don't follow you here.

Reply:
The `managedKafka` reference was a red herring. I was mostly just thinking of asserting that the properties we were interested in i.e. `key` & `value` were actually available rather than just inferring that from the fact the `uid` were different.